frontend/public/components/secret.jsx

```diff
@@ -16,7 +16,22 @@ data:
   username: YWRtaW4=
   password: MWYyZDFlMmU2N2Rm`);
 
-const menuActions = Cog.factory.common;
+const editInYaml = obj => {
+  if (obj.type === 'Opaque' && _.has(obj.data, 'WebHookSecretKey') && Object.keys(obj.data).length === 1) {
```

Comment:
I would use

```js
if (_.has(obj, 'data.WebHookSecretKey') && _.size(obj.data) === 1) {
```

The whole function could be simplified to

```js
// Edit in YAML if not editing a webhook secret with one key.
const editInYaml = obj => !_.has(obj, 'data.WebHookSecretKey') || _size(obj.data) !== 1;
```

We should make the string `WebHookSecretKey` a const defined once somewhere that we reuse to avoid typos.